src/resources.py

```diff
@@ -193,10 +199,11 @@ def _process_config_map(dest_folder, config_map, resource, unique_filenames, ena
             is_removed)
     if old_config_map.data is not None and not is_removed:
         for key in set(old_config_map.data.keys()) & set(config_map.data or {}):
-            old_config_map.data.pop(key)
+            if old_dest_folder == dest_folder:
```

Comment:
We leave only those keys (files) for removal which are not in the new resource anymore, entries not mentioned shall be removed. Your change disables this if the folder changed. So this means old files won't be deleted on destination folder changes. This seems to be an undesired behavior. I'd expect *all* files of the old configmap to be deleted if the destination folder changed, don't you agree?

Reply:
I'm not really sure what you mean.
Each key is popped out if it doesn't need to change (remove/create).
This condition makes sure that a key is a candidate for deletion only if the folder has changed! (it is not `is_removed`)
On the other hand - if the folder has changed -> you need to remove the original file (from old folder).

Reply:
`is_removed` relates to the full resource (either configmap or secret), not the individual keys.
So the whole branch is only reached if the resource has been changed or added.
To clean-up files which we created initially which are not valid anymore, we iterate on the intersection of the resource keys (the files which are not to be removed) and remove those keys from `old_config_map.data` and then iterate on the old map (with reduced keyset representing only the files which are to be removed) as a deletion (hard coded `True` for `is_deleted`).

The point is that if the destination folder changed, **all** "old" files need to be removed, not only the intersection of keys.

But maybe I am mixing things up here, the code grew quite a bit and would benefit from a refactoring. Your assertions should cover this case.

Reply:
You are correct, if the destination has changed - non of the keys will be popped out => therefor will be deleted.
The only thing I'd change is move the `key` loop inside the folder destination condition.
Refactoring sounds reasonable...

Reply:
Yes please, CI now also succeeded.

Reply:
Done.